client/ayon_core/modules/deadline/plugins/publish/collect_deadline_server_from_instance.py

```diff
@@ -17,6 +17,7 @@ class CollectDeadlineServerFromInstance(pyblish.api.InstancePlugin):
     label = "Deadline Webservice from the Instance"
     targets = ["local"]
     families = ["render",
+                "prerender",
```

Comment:
Or even better - can't we remove the families and just target all since instances without `farm=True` data get skipped anyway?

I'm personally now very curious looking at this large list which families we're trying to avoid hehe.

Reply:
100%. Out of scope of this convo but would they show up on the publish dialog or it will try run and then skip on runtime? It would be nice that the plugins listed on the publish process are only the ones that actually run so it's easier to debug what's going on. Would that require using `targets` instead or something that can be collected ahead of time?

Reply:
Plugins would still show up in Publisher with only `farm=True`, but they would do nothing. I prefer when they are not showing at all, that gets handled by `families`, not sure about `targets`

Reply:
Yeah I figured and I also prefer when they don't show up. @iLLiCiTiT what do you suggest?

Even with this limitation though I think it's better to remove the hard-coding of all the families as it's very hard to keep up to date and it's confusing

Reply:
Yup, targets isn't what this is for - that's more global for the whole current pyblish session - not per instance or per plug-in

I guess having a more global list of a "farm-targetable" families may be well worth it for now. Like have `lib.farm_families` or whatever. That is, if indeed, that list is shared across many plug-ins to keep in sync.

Or we swap around the problem and say that instances that are capable of farm submissions also get a `farm` family - and when it has then the plug-ins become relevant and show. However, the publisher UI is currently designed for instances to target one _product type_ which is the one defined by the creator instead of the creator being capable of already registering compatibility of the instance for multiple families so that the publisher UI would show that.

Reply:
I at least tried to synchronize all plugin's families.

This discussion here is valid and needed, but for separate issue.

Reply:
isn't it easy to just move that arg list somewhere else under the `ayon_core`?

Reply:
ayon_core.lib.farm_utils.py (for example):
```python
FARM_FAMILIES = [
    "arnold_rop",
    "image",
    "imagesequence",
    "karma_rop",
    "mantra_rop",
    "maxrender",
    "prerender.farm",
    "prerender.frames_farm",
    "prerender",
    "publish.hou",
    "redshift_rop",
    "render.farm",
    "render.frames_farm",
    "render",
    "renderFarm",
    "renderlayer",
    "usrender",
    "vray_rop",
    "vrayscene",
]
```
in the plugins
```python
from ayon_core.lib import farm_utils

...

families = farm_utils.FARM_FAMILIES
```

Reply:
to me it's a no-brainer doing it now